library/tasks/HISAT2.wdl

```diff
@@ -0,0 +1,301 @@
+  }
+
+  File hisat2_ref
+  File fq1
+  File fq2
+  String ref_name
+  String output_name
+  String sample_name
+
+  # runtime optional arguments
+  String? opt_docker
+  Int? opt_memory_gb
+  Int? opt_cpu
+  Int? opt_disk
+  Int? opt_preemptible
+
+  # runtime values
+  String docker = select_first([opt_docker, "quay.io/humancellatlas/secondary-analysis-hisat2:v0.2.2-2-2.1.0"])
+  Int machine_mem_mb = select_first([opt_memory_gb, 5]) * 1000
+  Int cpu = select_first([opt_cpu, 4])
+  # use provided disk number or dynmically size on our own, 10 is our zipped fastq -> bam conversion with 50GB of additional disk
+  Int disk = select_first([opt_disk, ceil((size(fq1, "GB") + size(fq2, "GB") * 10) + size(hisat2_ref, "GB") + 50)])
+  Int preemptible = select_first([opt_preemptible, 5])
+
+  parameter_meta {
+    hisat2_ref: ""
+    fq1: "gz forward fastq file"
+    fq2: "gz reverse fastq file"
+    ref_name: ""
+    output_name: ""
+    sample_name: ""
+    opt_docker: ""
+    opt_memory_gb: ""
+    opt_cpu: ""
+    opt_disk: ""
+    opt_preemptible: ""
+  }
+
+  command {
+    # Note that files MUST be gzipped or the module will not function properly
+    # This will be addressed in the future either by a change in how Hisat2 functions or a more
+    # robust test for compression type.
+
+    set -e
+
+    # fix names if necessary.
+    if [ "${fq1}" != *.fastq.gz ]; then
+        FQ1=${fq1}.fastq.gz
+        mv ${fq1} ${fq1}.fastq.gz
+    else
+        FQ1=${fq1}
+    fi
+    if [ "${fq2}" != *.fastq.gz ]; then
+        FQ2=${fq2}.fastq.gz
+        mv ${fq2} ${fq2}.fastq.gz
+    else
+        FQ2=${fq2}
+    fi
+
+   tar -zxvf "${hisat2_ref}"
+    hisat2 -t \
+      -x ${ref_name}/${ref_name} \
+      -1 $FQ1 \
+      -2 $FQ2 \
+      --rg-id=${sample_name} --rg SM:${sample_name} --rg LB:${sample_name} \
+      --rg PL:ILLUMINA --rg PU:${sample_name} \
+      --new-summary --summary-file ${output_name}.log \
+      --met-file ${output_name}.hisat2.met.txt --met 5 \
+      --seed 12345 \
+      -k 10 \
+      --secondary \
+      -p ${cpu} -S ${output_name}.sam
+    samtools sort -@ ${cpu} -O bam -o "${output_name}.bam" "${output_name}.sam"
+  }
+
+  runtime {
+    docker: docker
+    memory: machine_mem_mb + " MB"
+    disks: "local-disk " + disk + " HDD"
+    cpu: cpu
+    preemptible: preemptible
+  }
+
+  output {
+    File logfile = "${output_name}.log"
+    File metfile = "${output_name}.hisat2.met.txt"
+    File output_bam = "${output_name}.bam"
+  }
+}
+
+task HISAT2RSEM {
+  meta {
+    description: "JISHUUUUUUUUUUUUUUUU"
+  }
+
+  parameter_meta {
+    hisat2_ref: ""
+    fq1: "gz forward fastq file"
+    fq2: "gz reverse fastq file"
+    ref_name: ""
+    output_name: ""
+    sample_name: ""
+    docker: ""
+    memory_gb: ""
+    cpu: ""
+    disk: ""
+    preemptible: ""
+  }
+
+  File hisat2_ref
+  File fq1
+  File fq2
+  String ref_name
+  String output_name
+  String sample_name
+
+  # runtime values
```

Comment:
this version only has non optional values with the default values set in by default.  I like this because its way less boilerplate and it looks cleaner.  I think it could get a little confusing as to whether or not users will know they can/should overwrite these or not but I think if we put the right descriptions in the `parameter_meta` we might be able to make it more clear.

Reply:
For this version, should the `parameter_meta` contain the optional variables such as `opt_docker` instead of `docker`?

Reply:
it should not, i didnt do a good job at fleshing out the different version fully, will correct this 👍